### PR TITLE
[FIX] account,account_accountant: bank/cash related links

### PR DESCRIPTION
**Pull request by poma-odoo**
Before:
In the kanban card of Bank/Cash journals
 - Clicking the title of the journal opened recon widget (kanban) now it opens BSL list
 - Clicking on operations opened recon widget (kanban) now it opens BSL list
 - Journal Items menu was not visible out of debug mode, now it is visible
 - Journal Items view was opening recon widget (kanban) now it opens Journal Items In bank transactions tree view:
 - `Match` button was always visible, now it is visible if line is not matched, otherwise it renames to `View`

Related Enterprise PR:#https://github.com/odoo/enterprise/pull/33924

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/views/account_journal_dashboard_view.xml [hunk 1]
```diff
@@ -85,7 +85,7 @@
                                     <a role="menuitem" type="object" name="open_transfer_money">Internal Transfers</a>
                                 </div>
                                 <div>
-                                    <a role="menuitem" type="object" name="open_action" context="{'action_name': 'action_account_moves_all_a'}" groups="base.group_no_one">Journal Items</a>
+                                    <a role="menuitem" type="object" name="open_action" context="{'action_name': 'action_account_moves_all_a'}">Journal Items</a>
```
Comment (smetl) [inline addons/account/views/account_journal_dashboard_view.xml:88]: it should be 2 separated commits

File: odoo/addons/test_main_flows/static/tests/tours/main_flow.js [hunk 2]
```diff
@@ -890,11 +890,6 @@ tour.stepUtils.mobileModifier(tour.stepUtils.autoExpandMoreButtons('.o_control_p
     edition: "enterprise",
     trigger: "div.o_account_kanban div.o_kanban_card_header a.oe_kanban_action span:contains('Bank')",
     content: _t("Open the bank reconciliation widget"),
-}, {
-    mobile: false,
-    edition: "enterprise",
-    trigger: "button.o_switch_view.o_list",
-    content: _t("Move to the list view"),
```
Comment (smetl) [inline odoo/addons/test_main_flows/static/tests/tours/main_flow.js:897]: I don't understand what is wrong with that step. Creating from list or kanban is the same, right?

Comment (poma-odoo) [reply odoo/addons/test_main_flows/static/tests/tours/main_flow.js:897]: The order of the views are changed, clicking on journal name opens the tree view, so this step is not necessary and of course there is no `o_switch_view.o_list` to click on

**Comment by poma-odoo** [comment]
@smetl all done and rb is green

**Review by smetl — APPROVED** [review]

**Pull request closed without merging**